src/code/fault_ootn64.c

```diff
@@ -0,0 +1,835 @@
+                        x = 9;
+                        y = 1;
+                    } else {
+                        x = 0;
+                    }
+                    break;
+
+                case 0x9:
+                    if (btnPress == (BTN_A | BTN_B)) {
+                        x = 0xA;
+                    } else if (btnPress == BTN_A) {
+                        x = 0x5B;
+                    } else if (btnPress == BTN_B) {
+                        x = 0x5C;
+                    } else {
+                        x = 0;
+                    }
+                    break;
+
+                case 0x5B:
+                    if (btnPress == BTN_B) {
+                        x = 0xA;
+                    } else {
+                        x = 0;
+                    }
+                    break;
+
+                case 0x5C:
+                    if (btnPress == BTN_A) {
+                        x = 0xA;
+                    } else {
+                        x = 0;
+                    }
+                    break;
+
+                case 0xA:
+                    if ((btnCur == (BTN_A | BTN_B | BTN_START)) && (btnPress == BTN_START)) {
+                        f32 comboTimeS = OS_CYCLES_TO_USEC(osGetTime() - comboStartTime) / 1000000.0f;
+
+                        osSyncPrintf("入力時間 %f 秒\n", comboTimeS);
+                        if (comboTimeS <= 50.0f) {
+                            x = 0xB;
+                        } else {
+                            x = 0;
+                        }
+                    } else {
+                        x = 0;
+                    }
+                    break;
+
+                default:
+                    break;
+            }
+        }
+    }
+}
+
+extern s32 D_80105A90_unknown; // Arena_failcnt
+
+void func_800AF0E0(void) {
+    s32 i;
+
+    Fault_DrawRecBlack(22, 16, 276, 208);
+    Fault_DrawText(40, 30, "SegmentBaseAddress");
+
+    for (i = 0; i < ARRAY_COUNT(gSegments); i++) {
+        Fault_DrawText(40, 40 + 8 * i, "%2d:%08x", i, gSegments[i]);
+    }
+    Fault_DrawText(40, 180, "Arena_failcnt = %d", D_80105A90_unknown);
+}
+
+void Fault_DrawMemDumpContents(const char* title, void* memory, u32 arg2) {
+    s32 x;
+    s32 y;
+    u32* ptr = (u32*)((u32)memory & ~3);
+
+    Fault_DrawRecBlack(22, 16, 276, 208);
+    Fault_DrawText(36, 18, "%s %08x", title != NULL ? title : "PrintDump", ptr);
+
+    if ((ptr >= (u32*)0x80000000) && (ptr <= (u32*)0x80400000)) {
+        for (y = 28; y != 226; y += 9) {
+            Fault_DrawText(28, y, "%06x", ptr);
+            for (x = 82; x < 290; x += 52) {
+                Fault_DrawText(x, y, "%08x", *(ptr++));
+            }
+        }
+    }
+}
+
+void Fault_DrawMemDumpPC(OSThread* arg0) {
+    u32* pc = (u32*)arg0->context.pc;
+
+    if ((pc >= (u32*)0x80000020) && (pc <= (u32*)0x80400000)) {
+        Fault_DrawMemDumpContents("PC Trace", pc - 0x20, 0);
+    }
+}
+
+void Fault_DrawMemDumpSP(OSThread* arg0) {
```

Comment:
all these `OSThread* arg0` are called `OSThread* thread` in GC's fault.c (and some functions above already actually)